recipe_scrapers/kptncook.py

```diff
@@ -0,0 +1,121 @@
+                parsed_url = urlparse(recipe_request.url)
+            # Extract language from URL
+            query = parse_qs(parsed_url.query)
+            self.lang = query["lang"][0] if "lang" in query else KPTN_DEFAULT_LANGUAGE
+
+            # Build final recipe url (of type mobile.kptncook.com/*)
+            self.final_url = "".join(
+                ["https://", parsed_url.hostname, parsed_url.path, f"?lang={self.lang}"]
+            )
+            # Extract recipe id from the url path
+            recipe_uid = parsed_url.path.split("/")[-1]
+
+            # Request the final recipe json from the kptncook api
+            api_url = f"https://mobile.kptncook.com/recipes/search?kptnkey=6q7QNKy-oIgk-IMuWisJ-jfN7s6&lang={self.lang}"
+            json_request_body = [{"uid": recipe_uid}]
+            self.recipe_json = json.loads(
+                requests.post(api_url, headers=HEADERS, json=json_request_body).content
+            )[0]
+
+    @classmethod
+    def host(self, subdomain="mobile"):
+        return f"{subdomain}.kptncook.com"
+
+    def author(self):
+        author = self.recipe_json["authors"][0]
+        return f"{author['name']} ({author['link']})"
+
+    def title(self):
+        return self.recipe_json["title"]
+
+    def category(self):
+        return self.recipe_json["rtype"]
+
+    def total_time(self):
+        return self.recipe_json["cookingTime"] + self.recipe_json["preparationTime"]
+
+    def cook_time(self):
+        return self.recipe_json["cookingTime"]
+
+    def prep_time(self):
+        return self.recipe_json["preparationTime"]
+
+    def yields(self):
+        return 1
+
+    def nutrients(self):
+        return self.recipe_json["recipeNutrition"]
+
+    def canonical_url(self):
+        return self.final_url
+
+    def image(self):
+        return f"{self.recipe_json['imageList'][0]['url']}?kptnkey=6q7QNKy-oIgk-IMuWisJ-jfN7s6"
+
+    def ingredients(self):
+        return [
+            " ".join(
+                str(x)
+                for x in
+                # The filter is needed because "measure" and "quantity" fields are not always provided
+                filter(
+                    None,
+                    [
+                        ingredient.get("quantity"),
+                        ingredient.get("measure"),
+                        ingredient["ingredient"]["title"],
+                    ]
+                    if self.lang not in IMPERIAL_LANGUAGES
+                    else [
+                        ingredient.get("quantityUS"),
+                        ingredient.get("measureUS"),
+                        ingredient["ingredient"]["title"],
+                    ],
+                )
+            )
+            for ingredient in self.recipe_json["ingredients"]
+        ]
+
+    def instructions(self):
+        return "\n".join(step["title"] for step in self.recipe_json["steps"])
+
+    def ratings(self):
+        return ""
+
+    def cuisine(self):
+        return ""
```

Comment:
A small item of feedback related to these two fields:

For consistency let's return `None` instead of an empty string for fields that aren't provided by a website.

(reasoning: basically it's a [recommended practice](https://amir.rachum.com/blog/2012/08/25/you-cant-handle-the-truth/) in Python to check whether values are `None` (as opposed to falsy) -- similar to `null` vs falsy checks in JavaScript/TypeScript)